wp-includes/class-wp-service-workers.php

```diff
@@ -286,6 +286,116 @@ protected function get_base_script() {
 		return $script;
 	}
 
+	/**
+	 * Add precaching for wp-admin and wp-includes .js and .css files.
+	 */
+	public function precache_admin_assets() {
+
+		// Don't continue if debug mode.
+		if ( SCRIPT_DEBUG ) {
+			return;
+		}
+
+		do_action( 'admin_init' );
```

Comment:
Looks like doing `admin_init` here fixes the issues with the `colors` script and the `$_wp_admin_css_colors` not being set, not sure if that's the best solution.

Reply:
This is an issue that I had when wanting to `do_action('admin_enqueue_scripts')` for the admin as it is doing `wp_enqueue_scripts()` for the frontend. This fails, however, when the user is not logged-in and thus the service worker would not be able to be installed at `wp-login.php`, for example.

Reply:
It seems to work when testing, the only thing actually needed is for `$_wp_admin_css_colors` to be populated properly so we could just call `register_admin_color_schemes()` here instead, will replace it.